### ISSUE TITLE
Comparison operators for date fields don't work completely

### ISSUE THREAD

**Issue by mattsq** [ge-flight-analytics/Rems#3]
The following code works:
qry_select_past <- Rems::filter(qry_select, "'flight date' >= '2017-7-1'") %>%
                      Rems::filter("'flight date' < '2017-8-1'")
But this code:
qry_select_past <- Rems::filter(qry_select, "'flight date' > '2017-7-1'") %>%
                      Rems::filter("'flight date' <= '2017-8-1'")
Or:
qry_select_past <- Rems::filter(qry_select, "'flight date' >= '2017-7-1'") %>%
                      Rems::filter("'flight date' <= '2017-8-1'")
returns:
Error in datetime_filter(expr$op, fld_info, val_info) : 
  >: Unsupported conditional operator for discrete fields.
Error in datetime_filter(expr$op, fld_info, val_info) : 
  <=: Unsupported conditional operator for discrete fields.

I have Rems and Dplyr loaded.

**Comment by richardpineo** [comment]
This looks to me that the operators for datetime fields just aren't fully filled out.  The only comparison operators defined are < and >=.

from flt_query.R:
```
    date_ops <- list(
      "<" = "dateTimeBefore",
      ">=" = "dateTimeOnAfter"
    )
```

Ideally we should build out this R wrapper to support all of the date filter operators.
(from https://ems.efoqa.com/Docs/Rest/Resources)

Operation | Description
------------ | -------------
DateRelative | Represents the "relative" date operator.
DateSpecificMonths | Represents the "month range" date operator.
DateSpecificMonth | Represents the "specific month" date operator.
DateBeforeMonth | Represents the "before month" date operator.
DateOnAfterMonth | Represents the "on or after month" date operator.
DateSpecificDays | Represents the "day range" date operator.
DateSpecificDay | Represents the "specific day" date operator.
DateBeforeDay | Represents the "before day" date operator.
DateOnAfterDay | Represents the "on or after day" date operator.
DateTimeRange | Represents the "exact time range" date-time operator.
DateTimeBefore | Represents the "before exact time" date-time operator.
DateTimeOnAfter | Represents the "on or after exact time" date-time operator.
DateQuarterOfYear | Represents the "particular quarter of any year" date operator.
DateMonthOfYear | Represents the "particular month of any year" date operator.
DateDayOfWeek | Represents the "particular day of the week" date operator.
DateTimeHourOfDay | Represents the "particular hour of the day" date-time operator.

**Comment by mattsq** [comment]
Oh, I get what the issue is now. There actually isn't an equivalent in any of the operators for <= or > - there's only (Date|DateTime)Before and (Date|DateTime)OnAfter. If you wanted to implement the others using R, you'd have to do something hinky post-query I guess? Or add new operators in the API.

**Comment by richardpineo** [comment]
Yeah, I think you are right about this. One way to handle this might be to add a bit of code around:
https://github.com/ge-flight-analytics/Rems/blob/73a1e80d0a14e9897a7c64c2afcf3ef81624b463/R/flt_query.R#L527

I don't know R so the syntax below is bogus but the wrapper could make convenience functions to use the <= and > operators. There may also be a better way to handle it but this would be a short-term win to get at these operators.

```R
    date_ops <- list(
      "<" = "dateTimeBefore",
     "<=" = "dateTimeBefore",
      ">=" = "dateTimeOnAfter"
    )
   
   // If the operator is less-than-or-equal, then add a bit to 
   // the value and use the less-than operator.
   if( "op" ="<=" ) {
    valInfo += <one second>
  }
```

**Comment by mattsq** [comment]
My read of this SO answer is that you can increment POSIXCT directly in base R - +1 is equivalent to '+1 second'. So you could probably do Date <- Date + 1 or something cleaner.
https://stackoverflow.com/questions/50866366/adding-time-to-time-in-r

**Comment by richardpineo** [comment]
I'm going to leave this issue open since a better experience in the wrappers would be to have all the filter operations defined. (this is the big table i pasted above) To resolve this issue, I think we would add all the date operators to the wrapper.

However, the particular issue about >= defined but > not defined works as expected according to the specification in the EMS API filtering primitives. I think if we wanted to expand those filtering primitives then we could do so - but it would be better done in the API itself, not as a hack in the wrappers. So for that aspect of this issue, I'll add that to the backlog of the EMS API. My guess is that adding those would be pretty simple to do and would have reduced the confusion in this case.

**Comment by mattsq** [comment]
Comitted a change to at least make the error message slightly more informative! Now if you try to use an invalid operator, it'll tell you which ones are supported.
